### PR TITLE
Remove validations for noop instrument names and units 

### PR DESCRIPTION
**Pull request by breedx-splk**
We were chatting about API behavior and it was discovered that some metrics APIs do validation. We think that this is probably not what the spec intended, but also creates some log chatter and computational overhead in what should really be a true no-op implementation.

The [instrument naming spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule) does, oddly enough, impose naming restrictions, but these are implemented in the sdk. The api does not seem like place for this behavior.

The [metric api specification around units](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-unit) says that even the SDK isn't supposed to validate....so it makes little sense for the API to also be doing unit validations.

I think that it is unlikely that users are relying on this warning to be logged, so I believe that this change should be safe.

### CODE REVIEW

**Comment by breedx-splk** [comment]
@MrAlias @tsloughter FYI

**Comment by jack-berg** [comment]
> The [instrument naming spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule) does, oddly enough, impose naming restrictions, but these are implemented in the sdk. The api does not seem like place for this behavior.

FYI, [here's](https://github.com/open-telemetry/opentelemetry-java/pull/4383#discussion_r850898176) where that was originally added. Essentially, we did the thing we deemed most correct given the ambiguity in the specification about how API level argument restrictions should actually be enforced.

> The [metric api specification around units](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-unit) says that even the SDK isn't supposed to validate....so it makes little sense for the API to also be doing unit validations.

I think the spec is contradictory on this point. It needs to resolve the "the SDK is not expected to validate the unit of measurement" phrasing with the "It can have a maximum length of 63 characters". Not sure how these can coexist. I brought up this contradiction in [this issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2496#issuecomment-1114017967).

I'd want to hold off on merging this until the spec provides more clarity. It could end up being the case that noop API _is_ supposed to do this validation.

**Comment by MrAlias** [comment]
> I'd want to hold off on merging this until the spec provides more clarity. It could end up being the case that noop API _is_ supposed to do this validation.

This seems reasonable :+1:

**Comment by breedx-splk** [comment]
> I'd want to hold off on merging this until the spec provides more clarity. It could end up being the case that noop API _is_ supposed to do this validation.

Thanks, totally appreciate that perspective. I also commented on [the issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2496#issuecomment-1114017967) to try and round up some momentum.

**Comment by der-eismann** [comment]
Not sure if related, but with spring-security 6.0 we stumbled upon [this issue](https://github.com/spring-projects/spring-security/issues/12490#issuecomment-1410873702). While the space in the name is unwanted, keeping the names under 63 characters long seems to be hard with the given example of `spring.security.filterchains.SecurityContextHolderAwareRequestFilter.before`. Is this just about removing the log lines about the noop stuff that's currently polluting our logs?

**Comment by MrAlias** [comment]
Related specification PR: https://github.com/open-telemetry/opentelemetry-specification/pull/3171

**Comment by jack-berg** [comment]
@jkwatson FYI spec [PR #3171](https://github.com/open-telemetry/opentelemetry-specification/pull/3171) seems likely to merge and explicitly requires that noop implementations do not do any instrument name / unit validation. This is fine with me, since it makes the noop a true noop, but means that you won't be able to see the same name / unit warnings when the noop is installed versus the sdk.

**Review by jkwatson — APPROVED** [review]
🚢

File: api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java
Comment (jack-berg) [inline api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java:374]: These validation utils can be moved to the SDK now. `checkInstrumentName` probably makes most sense in `SdkMeter`. `checkValidInstrumentUnit` should be removed given that the spec has clarified that SDKs aren't supposed to do any validation on unit. You can do that removal in this PR if you want, or just move the method to `AbstractInstrumentBuilder`.

File: api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java
Comment (jack-berg) [inline api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java:67]: I think we should just delete `builder_InvalidName` and `builder_InvalidUnit`. It's weird to have tests floating around that don't relate to anything in the code.

**Comment by hriddhighosh4** [comment]
@breedx-splk Hi. It will be really helpful if we can get the changes in this PR merged.
We are using spring 3.0.2 and are getting the same error on the late opentelemtry collector contrib image.
It was fixed in spring security 6.0.2 but we are still getiing the error.

File: api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java
Comment (breedx-splk) [reply api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java:67]: @jack-berg Those seem fine to me, if not overly simple. They're really just testing around the noop builder, which I think I agree has limited testing value.

Do you want these removed as part of this PR? Are we waiting on that?

Comment (jack-berg) [reply api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java:67]: > They're really just testing around the noop builder

The coverage already exists below. The tests give the impression that there is some special condition about the specific instrument name / unit under test, which is no longer the case. 

Won't block on it, but I think they'd be confusing to future maintainers who aren't aware of the history. What's so special about `1` and `日` that warrants special tests?

File: api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java
Comment (jack-berg) [reply api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java:374]: Bumping this @breedx-splk.

File: api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java
Comment (breedx-splk) [reply api/all/src/test/java/io/opentelemetry/api/metrics/DefaultMeterTest.java:67]: That's cool then, I'll nuke 'em.

File: api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java
Comment (breedx-splk) [reply api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java:374]: Ok, I moved `checkValidInstrumentName()` to a new internal utility class in the sdk, rather than `SdkMeter`. This seemed a little more SRPy to me.

I did move the unit check over to the `AbstractInstrumentBuilder` class, and I think removal of that should be tackled in a separate PR.

Once those were moved out of the validator, the only things that were left in`ValidationUtil` were the logging methods, so I renamed it to `ApiUsageLogger`.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**
